crates/hir-ty/src/infer.rs

```diff
@@ -509,7 +509,7 @@ impl<'a> InferenceContext<'a> {
         }
     }
 
-    fn resolve_all(self) -> InferenceResult {
+    pub(crate) fn resolve_all(self) -> InferenceResult {
```

Comment:
this doesn't actually get called from anywhere new, does it?

Reply:
It is called in evaluating const generics. I'm probably doing something wrong here. The problem is, const eval is called inside type inference, but the constant in const generics share their body with their parent and type check at the same time, so in evaluating constants, the types inside of the const are not resolved yet. I made a clone of `InferenceContext` and called `resolve_all` on it as a hacky workaround.

Reply:
Can you put a FIXME on this/the call then?